x-pack/plugins/actions/server/sub_action_framework/types.ts

```diff
@@ -76,6 +77,36 @@ export type Validators<Config, Secrets> = Array<
   ConfigValidator<Config> | SecretsValidator<Secrets>
 >;
 
+export interface PreSaveConnectorHookParams<Config, Secrets> {
+  connectorId: string;
+  config: Config;
+  secrets: Secrets;
+  logger: Logger;
+  request: KibanaRequest;
+  services: HookServices;
+  isUpdate: boolean;
+}
+
+export interface PostSaveConnectorHookParams<Config, Secrets> {
+  connectorId: string;
+  config: Config;
+  secrets: Secrets;
+  logger: Logger;
+  request: KibanaRequest;
+  services: HookServices;
+  isUpdate: boolean;
+  wasSuccessful: boolean;
+}
+
+export interface PostDeleteConnectorHookParams<Config, Secrets> {
+  connectorId: string;
+  config: Config;
+  secrets: Secrets;
+  logger: Logger;
+  services: HookServices;
+  request: KibanaRequest;
+}
```

Comment:
Currently, the `post-delete` hook is not called if the DELETE fails - because the connector will still exist.

I think we definitely will need to add `wasSuccessful` if we add a `pre-delete` hook, so the `post-delete` could "clean up" anything the pre- hook did.